### PR TITLE
JCR-2469 : continue the indexation if a node doesn't have a primary type

### PR DESCRIPTION
**Pull request by waelghannem**
in order to continue indexing, we added a condition to check if the primary type property exists and return null if not.

### CODE REVIEW

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java [hunk 1]
```diff
@@ -940,7 +940,8 @@ protected IndexerChangesFilter initializeChangesFilter() throws RepositoryExcept
       }
       catch (InvocationTargetException e)
       {
-         throw new RepositoryException(e.getMessage(), e);
+         LOG.error("Unable to initialize change filter {}", changesFilterClassName, e);
```
Comment (thomasdelhomenie) [inline exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java]: Generally, an exception should be either loggued or rethrown, not both (see sonar rule for example https://rules.sonarsource.com/java/tag/error-handling/RSPEC-2139). If we need to log the error, can we log it in the caller method ?

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java [hunk 2]
```diff
@@ -629,13 +629,13 @@ public void start()
...
       {
-         LOG.error(e.getLocalizedMessage());
+         LOG.error(e.getLocalizedMessage(),e);
          handler = null;
          throw new RuntimeException(e.getLocalizedMessage(), e.getCause());
       }
       catch (RepositoryConfigurationException e)
       {
-         LOG.error(e.getLocalizedMessage());
+         LOG.error(e.getLocalizedMessage(),e);
```
Comment (thomasdelhomenie) [inline exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java]:
```suggestion
         LOG.error(e.getLocalizedMessage(), e);
```

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java [hunk 3]
```diff
@@ -629,13 +629,13 @@ public void start()
       }
       catch (RepositoryException e)
       {
-         LOG.error(e.getLocalizedMessage());
+         LOG.error(e.getLocalizedMessage(),e);
```
Comment (thomasdelhomenie) [inline exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java]:
```suggestion
         LOG.error(e.getLocalizedMessage(), e);
```

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java (hunk 2 above)
Comment (waelghannem) [reply exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java]: done.

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java (hunk 3 above)
Comment (waelghannem) [reply exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java]: done.

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java (hunk 1 above)
Comment (waelghannem) [reply exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java]: done.

Comment (thomasdelhomenie) [reply exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/SearchManager.java]: This comment must be considered for all the changes.

**Pull request merged**